core/txpool/legacypool/tx_overflowpool_test.go

```diff
@@ -157,6 +159,57 @@ func TestTxOverflowPoolHeapLen(t *testing.T) {
 	}
 }
 
+func TestTxOverflowPoolSlotCalculation(t *testing.T) {
+	// Initialize the pool with a maximum size of 2
+	pool := NewTxOverflowPoolHeap(2)
+
+	// Create two transactions with different slot requirements
+	tx1 := createTestTx(1, big.NewInt(1000)) // tx1 takes 1 slot
+	tx2 := createTestTx(2, big.NewInt(2000)) // tx2 takes 1 slot
+
+	// Add both transactions to fill the pool
+	pool.Add(tx1)
+	pool.Add(tx2)
+
+	if pool.Len() != 2 {
+		t.Fatalf("Expected pool size 2, but got %d", pool.Len())
+	}
+
+	dataSize := 40000
+	tx3 := createLargeTestTx(
+		3,                        // nonce
+		big.NewInt(100000000000), // gasPrice: 100 Gwei
+		dataSize,
+	) // takes 2 slots
+
+	// Create a third transaction with more slots than tx1
+	tx3Added := pool.Add(tx3)
+	assert.Equal(t, false, tx3Added)
+	assert.Equal(t, 2, pool.totalSize)
+
+	// Verify that the pool length remains at 2 after popping the oldest transaction
+	assert.Equal(t, 2, pool.Len(), "Expected pool size 2 after overflow")
+
+	// Attempt to add a duplicate transaction (tx3) to see if it increases currentSlotsUsed erroneously
+	initialTotalSize := pool.totalSize
+	pool.Add(tx3)
+
+	// The total size should not change since tx3 is already in the pool
```

Comment:
According to the logic, tx3 failed to be added in to overflowpool, right?

Reply:
Sorry it was bit outdated test. I just updated it.